src/Agent.Sdk/Knob/AgentKnobs.cs

```diff
@@ -512,11 +512,17 @@ public class AgentKnobs
             new EnvironmentKnobSource("AGENT_DISABLE_CLEAN_REPO_DEFAULT_VALUE"),
             new BuiltInDefaultKnobSource("false"));
 
-          public static readonly Knob IgnoreVSTSTaskLib = new Knob(
+        public static readonly Knob IgnoreVSTSTaskLib = new Knob(
             nameof(IgnoreVSTSTaskLib),
             "Ignores the VSTSTaskLib folder when copying tasks.",
             new RuntimeKnobSource("AZP_AGENT_IGNORE_VSTSTASKLIB"),
             new EnvironmentKnobSource("AZP_AGENT_IGNORE_VSTSTASKLIB"),
             new BuiltInDefaultKnobSource("false"));
+
+        public static readonly Knob CheckForTaskDeprecation = new Knob(
+            nameof(CheckForTaskDeprecation),
+            "If true, the agent will check in the 'Initialize job' step each task used in the job for task deprecation.",
+            new EnvironmentKnobSource("AZP_AGENT_CHECK_FOR_TASK_DEPRECATION"),
+            new BuiltInDefaultKnobSource("false"));
     }
```

Comment:
RuntimeKnobSource needed for feature flag?

Reply:
`RuntimeKnobSource` is not needed for FF since it is `false` by default and will be enabled on the ADO server side via the TFS CC release.